### ISSUE TITLE
older maps breaking on new chroma versions

### ISSUE THREAD

**Issue by Aldin101** [Aeroluna/Chroma#34]
Older maps do not work like they used to (bad apple by peecore and nulcrtl by reactx ect.) do not work like they used to. I have tested them with game version 1.13.2 (resetting my game save while doing so) and they worked as expected.

**Comment by homebrewuser** [comment]
huh... same here

**Comment by Aeroluna** [comment]
Please describe what you mean by "do not work like they used to", and also provide log files please

**Comment by Fatalution** [comment]
> Older maps do not work like they used to (bad apple by peecore and nulcrtl by reactx ect.) do not work like they used to. I have tested them with game version 1.13.2 (resetting my game save while doing so) and they worked as expected.

Old 1.13.2 and older maps' Chroma effects won't display correctly in new version of the game. That's because Chroma was based on the prop ID system and prop IDs were changed (and changed again) during 1.13.4 and 1.16 update.

Since 1.13.4, Chroma is using new unified lightID system with translation table. That means that there is a translation layer happening from Chroma's lightID to new game's lightID and therefore the effects can't break anymore (considering the light source won;t be removed, in that case the effect on that lightID will just not display).

In short, that's expected. New maps that got released since 1.13.4 and up are not affected of this issue

**Comment by Aldin101** [comment]
You did not close this issue making me think that there will be a way to fix this issue. Is there and if so, will you implement it?
________________________________

**Comment by Fatalution** [comment]
I'm not the Chroma's dev, that would be @Aeroluna . However, I think he will answer you in the same fashion.

Sadly, there is no way to fix older maps. Doing so would require translation table for every version of the game and translating every light and propID for every environment. Moreover, the details of environments have changed (for instance, Default in 1.13.2 didn't have pillars which are now a part of lasers group) - imagine how ludicrous that would be.

So, the answer is 99% no and it is not possible. That's why Chroma switched to using unified lightID system in the first place
